cmd/aws-janitor/main.go

```diff
@@ -34,16 +40,30 @@ import (
 )
 
 var (
-	maxTTL    = flag.Duration("ttl", 24*time.Hour, "Maximum time before attempting to delete a resource. Set to 0s to nuke all non-default resources.")
-	region    = flag.String("region", "", "The region to clean (otherwise defaults to all regions)")
-	path      = flag.String("path", "", "S3 path for mark data (required when -all=false)")
-	cleanAll  = flag.Bool("all", false, "Clean all resources (ignores -path)")
-	logLevel  = flag.String("log-level", "info", fmt.Sprintf("Log level is one of %v.", logrus.AllLevels))
-	dryRun    = flag.Bool("dry-run", false, "If set, don't delete any resources, only log what would be done")
-	ttlTagKey = flag.String("ttl-tag-key", "", "If set, allow resources to use a tag with this key to override TTL")
+	maxTTL      = flag.Duration("ttl", 24*time.Hour, "Maximum time before attempting to delete a resource. Set to 0s to nuke all non-default resources.")
+	region      = flag.String("region", "", "The region to clean (otherwise defaults to all regions)")
+	path        = flag.String("path", "", "S3 path for mark data (required when -all=false)")
+	cleanAll    = flag.Bool("all", false, "Clean all resources (ignores -path)")
+	logLevel    = flag.String("log-level", "info", fmt.Sprintf("Log level is one of %v.", logrus.AllLevels))
+	dryRun      = flag.Bool("dry-run", false, "If set, don't delete any resources, only log what would be done")
+	ttlTagKey   = flag.String("ttl-tag-key", "", "If set, allow resources to use a tag with this key to override TTL")
+	pushGateway = flag.String("push-gateway", "", "If specified, push prometheus metrics to this endpoint.")
 
 	excludeTags common.CommaSeparatedStrings
 	includeTags common.CommaSeparatedStrings
+
+	sweepCount int
+
+	cleaningTimeHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:        "aws_janitor_job_duration_time_seconds",
+		ConstLabels: prometheus.Labels{},
+		Buckets:     prometheus.ExponentialBuckets(1, 1.4, 30),
+	}, []string{"type", "status", "region"})
+
+	sweepCountGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name:        "aws_janitor_sweeps",
```

Comment:
nit: maybe `aws_janitor_swept_resources`? `sweeps` makes me think of how many times the sweep operation was run, rather than the number of items swept.